chapters/api_concepts.txt

```diff
@@ -242,29 +242,36 @@ brevity:
 [[api_concepts_introspection]]
 === Object Introspection
 
-ANARI supports introspection of objects, i.e., querying supported object
+ANARI supports introspection of objects, i.e., querying supported features, object
 subtypes and information about their parameters, which is particularly useful
 to enumerate and inspect implementation-specific object features. The intended
 use is to allow for building a graphical user interface for, e.g., specific
 renderers or materials, and to provide hints for the debug layer to aid in
-validating features.  Because the meta-information of objects and parameters
-supported by devices is static, it can be queried without creating a device
-first by passing the device name.
+validating features.
+
+Device subtypes and their implemented features can be queried
+from a library before instantiating a device. Object and parameter specific
+information can be queried from an instantiated device object.
 
 [source,cpp]
 ----
 const char **anariGetDeviceSubtypes(ANARILibrary);
+const char **anariGetDeviceFeatures(ANARILibrary, const char *deviceSubtype);
 ----
 
-To get a list of device subtypes implemented in a library, call function
+A list of device subtypes implemented in a library is retrieved by calling
 `anariGetDeviceSubtypes`. It returns `NULL` if there are no devices, or a
 `NULL`-terminated list of `0`-terminated C-strings with the names of the
 devices. The first (if any) device is the default device.
 
+The list features implemented by a device is retrieved by calling
+`anariGetDeviceFeatures` with a device subtype returned by
+`anariGetDeviceSubtypes`. It returns a `NULL`-terminated list of `0`-terminated
+C-strings with the names of the implemented features.
+
 [source,cpp]
 ----
-const char **anariGetObjectSubtypes(
-    ANARILibrary, const char *deviceSubtype, ANARIDataType objectType);
+const char **anariGetObjectSubtypes(ANARIDevice, ANARIDataType objectType);
```

Comment:
Do we still need that function? It was also introduced before feature queries: for every object subtype we have a feature string (which can be queried via `anariGetDeviceFeatures`). The only difference is to here is to *only* get the subtypes, and directly the names.

Reply:
I think it's better to keep this function so that you have a definitive list of strings to be passed through the API, as opposed to needing to parse feature strings and hope that they line up -- even if that were guaranteed with official extensions, it would be impossible to enforce that convention for vendor extensions.

Reply:
Particularly with renderer names you probably want this. It makes sense for those to be "discoverable" at runtime without knowledge of device specific features.